### PR TITLE
Add update workspace page

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description

Add update workspace page

## Screenshot

image
image

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_overview.tsx [hunk 1]
```diff
@@ -7,25 +7,43 @@ import React from 'react';
...
+    if (!currentWorkspace || !currentWorkspace.id) {
+      notifications?.toasts.addDanger({
+        title: i18n.translate('Cannot find current workspace', {
+          defaultMessage: 'Cannot update workspace',
+        }),
+      });
+      return;
+    }
+    application.navigateToApp(WORKSPACE_APP_ID, {
+      path: PATHS.update + '?' + WORKSPACE_ID_IN_SESSION_STORAGE + '=' + currentWorkspace.id,
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_overview.tsx:36]: It is recommend to use `formatUrlWithWorkspaceId` method workspaces client provides to generate the final url.

File: src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx
Comment (yuye-aws) [inline src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx:1]: This import path can be shortened to:
`import { useOpenSearchDashboards } from '../../../opensearch_dashboards_react/public';`

File: src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { EuiPage, EuiPageBody, EuiPageHeader, EuiPageContent } from '@elastic/eui';
+import { useObservable } from 'react-use';
+import { i18n } from '@osd/i18n';
+import { of } from 'rxjs';
+
+import { WorkspaceAttribute } from 'opensearch-dashboards/public';
+import { useOpenSearchDashboards } from '../../../../../../src/plugins/opensearch_dashboards_react/public';
```
Comment (yuye-aws) [inline src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx:13]: This import path can be shortened to:
`import { useOpenSearchDashboards } from '../../../opensearch_dashboards_react/public';`

File: src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx [hunk 3]
```diff
@@ -62,8 +63,14 @@ interface WorkspaceFormProps {
   application: ApplicationStart;
   onSubmit?: (formData: WorkspaceFormData) => void;
   defaultValues?: WorkspaceFormData;
+  opType?: string;
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx:66]: How about update `opType?: string` to `opType?: WORKSPACE_OP_TYPE_CREATE | WORKSPACE_OP_TYPE_UPDATE `

File: src/plugins/workspace/public/components/workspace_overview.tsx [hunk 4]
```diff
@@ -7,25 +7,43 @@ import React from 'react';
...
-  } = useOpenSearchDashboards();
+    services: { workspaces, application, notifications },
+  } = useOpenSearchDashboards<{ application: ApplicationStart }>();
 
   const currentWorkspace = useObservable(
     workspaces ? workspaces.client.currentWorkspace$ : of(null)
   );
 
+  const onUpdateWorkspaceClick = () => {
+    if (!currentWorkspace || !currentWorkspace.id) {
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_overview.tsx:27]: Could we change to `!currentWorkspace?.id`?

File: src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+  const currentWorkspace = useObservable(
+    workspaces ? workspaces.client.currentWorkspace$ : of(null)
+  );
+
+  const excludedAttribute = 'id';
+  const { [excludedAttribute]: removedProperty, ...otherAttributes } =
+    currentWorkspace || ({} as WorkspaceAttribute);
+
+  const [currentWorkspaceFormData, setCurrentWorkspaceFormData] = useState<
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx:37]: The `useState` will cause extra re-render after `setCurrentWorkspaceFormData` called. How about change to `useRef`?

**Pull request merged**